libpod/image/image.go

```diff
@@ -567,3 +567,63 @@ func Import(path, reference string, writer io.Writer, signingOptions SigningOpti
 	}
 	return runtime.NewFromLocal(reference)
 }
+
+// MatchRepoTag takes a string and tries to match it against an
+// image's repotags
+func (i *Image) MatchRepoTag(input string) (string, error) {
+	var maxKey string
+	// first check if we have an exact match with the input
+	if util.StringInSlice(input, i.Names()) {
+		return input, nil
+	}
+
+	results := make(map[string]int)
+
+	// next check if we are missing the tag
+	dcImage, err := decompose(input)
+	if err != nil {
+		return "", err
+	}
+
+	for _, repoName := range i.Names() {
+		count := 0
+		dcRepoName, err := decompose(repoName)
+		if err != nil {
+			return "", err
+		}
+		if dcRepoName.registry == dcImage.registry && dcImage.registry != "" {
+			count++
+		}
+		if dcRepoName.name == dcImage.name && dcImage.name != "" {
+			count++
+		} else if splitString(dcRepoName.name) == splitString(dcImage.name) {
+			count++
+		}
+		if dcRepoName.tag == dcImage.tag {
+			count++
+		}
+		results[repoName] = count
+	}
+	maxValue := -1
```

Comment:
this whole section sucks ... i would love some suggestions on how to deal with this.  i have a map with the name of the tag pointing to an int ... the higher the int, the more likely the match is.  We've already used the user input to get the Image, so there needs to be a match.  ideally id somehow get the max and make sure max isnt duplicated  (should never be).  @mheon @rhatdan @vrothberg